// ngx-wig/esm2020/lib/ngx-wig.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxWigComponent } from './ngx-wig.component';
import { FormsModule } from '@angular/forms';
import { DEFAULT_LIBRARY_BUTTONS, BUTTONS } from './config';
import * as i0 from "@angular/core";
export function getWindowObject() {
    return window;
}
export class NgxWigModule {
    static forRoot(config) {
        return {
            ngModule: NgxWigModule,
            providers: [
                {
                    provide: BUTTONS,
                    multi: true, useValue: ((!config || !config?.buttonsConfig) ? DEFAULT_LIBRARY_BUTTONS : config?.buttonsConfig)
                },
                { provide: 'WINDOW', useFactory: getWindowObject },
            ],
        };
    }
    static forChild() {
        return { ngModule: NgxWigModule };
    }
}
NgxWigModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgxWigModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxWigModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgxWigModule, declarations: [NgxWigComponent], imports: [CommonModule,
        FormsModule], exports: [NgxWigComponent] });
NgxWigModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgxWigModule, providers: [
        { provide: BUTTONS, multi: true, useValue: DEFAULT_LIBRARY_BUTTONS },
        { provide: 'WINDOW', useFactory: getWindowObject },
    ], imports: [[
            CommonModule,
            FormsModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgxWigModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgxWigComponent
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                    ],
                    exports: [NgxWigComponent],
                    providers: [
                        { provide: BUTTONS, multi: true, useValue: DEFAULT_LIBRARY_BUTTONS },
                        { provide: 'WINDOW', useFactory: getWindowObject },
                    ]
                }]
        }] });
